### ISSUE TITLE
CTPPSPixelDataFormatter:  CTPPSPixelRawToDigi:hltCTPPSPixelDigis unphysical col and/or row  ...

### ISSUE THREAD

**Issue by silviodonato** [cms-sw/cmssw#37483]
Running the latest HLT menu in Run-2 data and `12_4_0_pre2` gives a lot of warnings from PPS

```
hltGetConfiguration /dev/CMSSW_12_3_0/GRun \
    --globaltag auto:run3_hlt \
    --data \
    --unprescale \
    --eras Run2_2018 \
    --output minimal \
    --max-events 100 \
    --l1-emulator Full --l1 L1Menu_Collisions2022_v1_0_0_xml \
    --customise HLTrigger/Configuration/customizeHLTforCMSSW.customiseFor2018Input,HLTrigger/Configuration/customizeHLTforPatatrack.customizeHLTforPatatrackTriplets \
    --input file:/eos/cms/store/data/Run2018D/EphemeralHLTPhysics2/RAW/v1/000/323/775/00000/17ADD12B-52E2-8C4C-B375-8AF943A24212.root \
    > hltData.py
```

gives 2519 times a warning like
```
%MSG-e CTPPSPixelDataFormatter:  CTPPSPixelRawToDigi:hltCTPPSPixelDigis  07-Apr-2022 10:36:41 CEST Run: 323775 Event: 178905954
 unphysical col and/or row  nllink=35 nroc=1 adc=36 col=50 row=92
%MSG
```
I ran over only 100 events.

I know that the PPS detector underwent some big changes between run-2 and run-3 data, so probably the warning is "correct", but it is very annoying to get so many warnings.

**Comment by silviodonato** [comment]
assign @cms-sw/ctpps-dpg-l2 
cc @cms-sw/hlt-l2

**Comment by silviodonato** [comment]
A "minimal" configuration to reproduce the warning is
```
cd CMSSW_12_4_0_pre2/src
cmsenv
hltGetConfiguration /dev/CMSSW_12_3_0/GRun \
    --globaltag auto:run3_hlt \
    --eras Run2_2018 \
    --output none \
    --max-events 1 \
    --customise HLTrigger/Configuration/customizeHLTforCMSSW.customiseFor2018Input \
    --input file:/eos/cms/store/data/Run2018D/EphemeralHLTPhysics2/RAW/v1/000/323/775/00000/17ADD12B-52E2-8C4C-B375-8AF943A24212.root \
    --path HLT_PPSMaxTracksPerRP4_v1 \
    --open \
    > hltData.py

cmsRun hltData.py >& log 
```

**Comment by makortel** [comment]
assign reconstruction

**Comment by fabferro** [comment]
I can see this fragment in HLT configuration code

fragment.hltCTPPSPixelDigis = cms.EDProducer( "CTPPSPixelRawToDigi",
isRun3 = cms.bool( True ),
includeErrors = cms.bool( True ),
inputLabel = cms.InputTag( "rawDataCollector" ),
mappingLabel = cms.string( "RPix" )
)

If this is run with the flag isRun3=True, then the data unpacker uses Run3 code. That can generates the error messages you see.
This does not happen in "normal" (no HLT) code processing because this flag is governed via era mechanism.

**Comment by missirol** [comment]
@fabferro , thanks for the explanation.

@silviodonato , this looks a case that should be handled in the HLT customisation `customiseFor2018Input`. What do you think?

Looks like we can unassign RECO, and assign HLT.

**Comment by missirol** [comment]
unassign reconstruction

**Comment by missirol** [comment]
assign hlt

**Comment by silviodonato** [comment]
> @silviodonato , this looks a case that should be handled in the HLT customisation `customiseFor2018Input`. What do you think?

yes, indeed. We might also use our HLTrigger.Configuration.Eras import modifyHLTforEras

**Comment by missirol** [comment]
A solution is in https://github.com/cms-sw/cmssw/pull/37518 (to be backported to `12_3_X`, unless there are objections).

> yes, indeed. We might also use our HLTrigger.Configuration.Eras import modifyHLTforEras

To follow what is currently done for other parts of the reconstruction (e.g. pixel reco), in #37518 I went for a simple customisation function without touching `modifyHLTforEras`.

**Comment by missirol** [comment]
+hlt

 - fixed in 12_3_X (#37521) and 12_4_X (#37518)
 
 @silviodonato , feel free to close the issue if you think the problem is solved.
